src/Resources/contao/classes/FrontendUser.php

```diff
@@ -167,57 +190,30 @@ public function authenticate()
 	 * Add the auto login resources
 	 *
 	 * @return boolean
+	 *
+	 * @deprecated Deprecated since Contao 4.5, to be removed in Contao 5.0.
+	 *             Use the security.interactive_login event instead.
 	 */
 	public function login()
 	{
-		// Default routine
-		if (parent::login() == false)
-		{
-			return false;
-		}
+		@trigger_error('Using User::FrontendUser() has been deprecated and will no longer work in Contao 5.0. Use the security.interactive_login event instead.', E_USER_DEPRECATED);
 
-		// Set the auto login data
-		if (\Config::get('autologin') > 0 && \Input::post('autologin'))
-		{
-			$time = time();
-			$strToken = md5(uniqid(mt_rand(), true));
-
-			$this->createdOn = $time;
-			$this->autologin = $strToken;
-			$this->save();
-
-			$this->setCookie('FE_AUTO_LOGIN', $strToken, ($time + \Config::get('autologin')), null, null, \Environment::get('ssl'), true);
-		}
-
-		return true;
+		return false;
```

Comment:
We are returning `parent::logout()` in the `logout()` method but `false` here. Should we return `parent::login()` instead?

Reply:
Yes please!